packages/react-form/src/hooks/submit.ts

```diff
@@ -16,24 +16,21 @@ export function useSubmit<T extends FieldBag>(
   const [submitting, setSubmitting] = useState(false);
   const [remoteErrors, setRemoteErrors] = useState([] as FormError[]);
 
-  const submit = useCallback(
-    async (event?: React.FormEvent) => {
-      if (event && event.preventDefault && !event.defaultPrevented) {
-        event.preventDefault();
-      }
+  async function submit(event?: React.FormEvent) {
```

Comment:
Rerendering a `Form` element is not expensive, so this being reference equal between renders is not a big deal. In addition, checking `fieldBag` meant that in most cases this would be regenerated each render anyway.

Reply:
I am not sure that it is safe to say that the thing this callback gets passed to is cheap to re-render, in the end you really have no idea what the cost is. Ideally we would memoize because the consumer can't, but if that's impossible, then that's fine.